src/streams/adapters/EventedStreamSource.ts

```diff
@@ -0,0 +1,50 @@
+import Evented from '../../Evented';
+import { Handle } from '../../interfaces';
+import on, { EventEmitter, EventTarget, ExtensionEvent } from '../../on';
+import Promise from '../../Promise';
+import { Source } from '../ReadableStream';
+import ReadableStreamController from '../ReadableStreamController';
+
+type EventTargetTypes = Evented | EventEmitter | EventTarget;
+type EventTypes = Array<ExtensionEvent | string> | ExtensionEvent | string;
+
+export default class EventedStreamSource implements Source<Event> {
+	protected _controller: ReadableStreamController<Event>;
+	protected _target: EventTargetTypes;
+	protected _events: Array<ExtensionEvent | string>;
```

Comment:
Why not make a type alias of `type EventNames = ExtensionEvent | string` and then use `EventNames[]`?